### PR TITLE
Added Open Source Note functionality for Timeline icons

### PR DESCRIPTION
**Pull request by danlee1025**
_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._

Addresses JIRA-1838.

-  Imported `VisualizerMenu` into `TimelineEventsVisualizer`.
-  Included source information in `TimelineSection`.
-  Added `sourceClinicalNote` data to some `ProcedureRequested` and `MedicationRequested` entries in Ihanos JSON.
    -  I did not include this data in `demo1` and `demo2` but new progression values should have the open source note functionality.

## Submitter:

**Running the Application**

- [x] Manually tested in Chrome (primary browser for testing)
- [x] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [x] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [x] Branch implements task as expected
-  Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [x] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
-  Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
-  Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
-  Document any new APIs/extension points
-  Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [x] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [x] Code is commented

**Tests**

- [x] Existing tests passed
-  Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

**Comment by sgober** [comment]
This looks great Dan!

[SO1] This isn't an issue with what you did since it is on fluxnotes.org, but if you do a stop medication for imatinib, the timeline doesn't update to stop that medication.  Is this intended behavior?

**Comment by danlee1025** [comment]
I looked at SO1 quickly and we should probably make this a JIRA.  It looks like when we do `#stop medication` it doesn't set the `endDate` for the medication.  It's pretty trivial to set the `endDate` but we also have to make sure to save the old `endDate` so that it can be restored if the shortcut is deleted.

**Comment by sgober** [comment]
@danlee1025 that works for me.  Can you create it since you looked into it more?

**Comment by danlee1025** [comment]
For DP1, the 2 `ProcedureRequested` dates I changed to match up with the date of the source note that was opening.  It seemed like the Progression and Medication entry dates matched up with a source note so I made that change for the Procedure entries too.

**Comment by gregquinn2001** [comment]
We have to be real careful when changing data on pilot1 patient. Let me take a look. That patient has had the most clinical input and is the most clinically complete!

File: src/dataaccess/GistAdjuvantIhanos.json [hunk 1]
```diff
@@ -965,6 +972,13 @@
                     }
                 }
             }
+        },
+        "SourceClinicalNote": {
+            "_ShrId": "788dcbc3-ed18-470c-89ef-35ff91854c7f",
+            "_EntryId": "5110",
```
Comment (gregquinn2001) [inline src/dataaccess/GistAdjuvantIhanos.json]: Note 5110 is to order a CT-guided biopsy but this procedure requested is for the endoscopic biopsy. Also, I think we have some confusion caused by a misuse of mCODE entries a little.

Ihanos had 2 biopsies: CT-guided scheduled to be performed on 3 August 2016 and endoscopic scheduled to be performed on 18 July 2016. The notes where those are ordered (in PLAN section), should have the same date as the creation date of the procedure requested. We should have notes corresponding to the day they were performed as well that will be source for procedure performed but we have a separate JIRA to address that I believe.

File: src/dataaccess/GistAdjuvantIhanos.json [hunk 2]
```diff
@@ -1132,6 +1146,13 @@
                     }
                 }
             }
+        },
+        "SourceClinicalNote": {
+            "_ShrId": "788dcbc3-ed18-470c-89ef-35ff91854c7f",
+            "_EntryId": "5160",
```
Comment (gregquinn2001) [inline src/dataaccess/GistAdjuvantIhanos.json]: This is actually the note that is requesting the surgery so it should be the source node for the Procedure Performed for the surgery not the CT-guided biopsy

File: src/dataaccess/GistAdjuvantIhanos.json [hunk 3]
```diff
@@ -1299,6 +1320,13 @@
                     }
                 }
             }
+        },
+        "SourceClinicalNote": {
+            "_ShrId": "788dcbc3-ed18-470c-89ef-35ff91854c7f",
+            "_EntryId": "5200",
```
Comment (gregquinn2001) [inline src/dataaccess/GistAdjuvantIhanos.json]: Note 5200 is the surgical note and would be linked to the ProcedurePerformed for the surgery. The ProcedureRequested for the surgery should be linked to note 5160

File: src/dataaccess/GistAdjuvantIhanos.json [hunk 4]
```diff
@@ -8489,6 +8517,13 @@
                     }
                 }
             }
+        },
+        "SourceClinicalNote": {
+            "_ShrId": "788dcbc3-ed18-470c-89ef-35ff91854c7f",
+            "_EntryId": "5310",
```
Comment (gregquinn2001) [inline src/dataaccess/GistAdjuvantIhanos.json]: Good catch!

File: src/dataaccess/GistAdjuvantIhanos.json [hunk 5]
```diff
@@ -9301,6 +9336,13 @@
                     }
                 }
             }
+        },
+        "SourceClinicalNote": {
+            "_ShrId": "788dcbc3-ed18-470c-89ef-35ff91854c7f",
+            "_EntryId": "5410",
```
Comment (gregquinn2001) [inline src/dataaccess/GistAdjuvantIhanos.json]: this looks good too!

File: src/dataaccess/GistAdjuvantIhanos.json (hunk 3 above)
Comment (danlee1025) [reply src/dataaccess/GistAdjuvantIhanos.json]: Ok I will make this change.  

- ProcedureRequested for surgery -> note 5160
- ProcedureRequested for CT-guided Biopsy -> note 5110
- ProcedureRequested for Endoscopic Biopsy -> note 5000?(Not really sure what "Order EGD with EUS" means)

**Comment by danlee1025** [comment]
DP1 should be addressed.  Spoke with Greg and updated the `ProcedureRequested` dates and linked to the correct source notes.

**Comment by Dtphelan1** [comment]
With DP1 addressed and Greg's comments addressed I feel good! Everything still works as I expected before. Great work Dan!

**Pull request merged**